bokehjs/src/lib/models/tools/button_tool.ts

```diff
@@ -20,7 +21,12 @@ export abstract class ButtonToolButtonView extends DOMView {
 
   render(): void {
     empty(this.el)
-    this.el.classList.add(this.model.icon)
+    if (isString(this.model.icon)) {
+      if (this.model.icon.startsWith("bk-tool-icon-"))
+        this.el.classList.add(this.model.icon)
+      else
+        this.el.style.backgroundImage = "url('data:image/png;base64," + this.model.icon + "')"
```

Comment:
Just occurs to me this is problematic, as ostensibly `Image` will accept other valid PIL image formats. But I'm not sure data URI's can actually be more than PNGs. Might end up simply restricting what `Image` accepts.